### PR TITLE
Add payload to new version of spend tx

### PR DESCRIPTION
**Pull request by lucafavatella**
Supersedes PR #1032

TODO:
* [x] Check Python UAT tests pass
* [x] Check unit tests pass
* [x] Check `make test` passes
* [x] Check Dialyzer passes
* [ ] Check aest_hard_fork_SUITE does not fail, and adapt it if it fails
  * Cannot check as building :local Docker image fails with g++
* [ ] Cover spend tx versions in system test in aest_hard_fork_SUITE
  * I suggest to do this in subsequent PR
* [ ] Ensure release notes updated
  * I suggest to do this in separate PR

### CODE REVIEW

**Comment by lucafavatella** [comment]
I am performing a few local checks then I plan to remove the WIP label.

**Comment by gorillainduction** [comment]
I'm guessing that we will revert at least some of these changes at some point before main net. It seems bad to start out with already deprecated protocol changes.

File: apps/aecore/src/aec_spend_tx.erl [hunk 1]
```diff
@@ -90,7 +126,10 @@ check(#spend_tx{recipient = RecipientPubkey} = SpendTx, _Context, Trees0, Height
             end;
         {error, _Reason} = Error ->
             Error
-    end.
+    end;
+check(#spend_tx{vsn = Vsn} = SpendTx, _Context, Trees0, Height, ConsensusVersion)
```
Comment (jur0) [inline apps/aecore/src/aec_spend_tx.erl]:
```
apps/aecore/src/aec_spend_tx.erl:130: Warning: variable 'Height' is unused
apps/aecore/src/aec_spend_tx.erl:130: Warning: variable 'SpendTx' is unused
apps/aecore/src/aec_spend_tx.erl:130: Warning: variable 'Trees0' is unused
```

Comment (lucafavatella) [reply apps/aecore/src/aec_spend_tx.erl]: Addressed in new commit

**Comment by lucafavatella** [comment]
CI test job still running for warnings removal. Though I saw tests passing with warning https://circleci.com/gh/aeternity/epoch/9349 that is the same. Merging.

**Comment by lucafavatella** [comment]
Ticket https://www.pivotaltracker.com/story/show/156942570

**Pull request merged**